### ISSUE TITLE
Usability: Pipeline editor: "legacy" runtime images are not displayed

### ISSUE THREAD

**Issue by ptitzler** [elyra-ai/elyra#946]
1.3-dev

- Clone the Elyra examples repository https://github.com/elyra-ai/examples.git
- Navigate to pipelines/dax_noaa_weather_data
- Open the pre-configured pipeline. It was created using Elyra 1.2.x
- Hover over a node. Note the associated runtime image

   ![image](https://user-images.githubusercontent.com/13068832/94731032-ffc2ff80-0318-11eb-9ffd-ec627fb2c080.png)

- Open the properties of the selected node. Note that the selected runtime image drop-down is empty

   ![image](https://user-images.githubusercontent.com/13068832/94730928-d73b0580-0318-11eb-9726-2e301037d5f6.png)

It appears that this will always happen when Elyra is re-installed and new runtime image versions are included. In my case the pipeline was created using `pandas 1.0.3` whereas 1.3-dev includes only a definition for `pandas 1.1.1`. 

While the behavior is expected from an implementation point of view it is confusing to a user because there are now two scenarios for which the entry is empty.  
 - A user never assigned a runtime image to a node (leads to validation error)
 - A user assigned a runtime image to a node but the runtime image configuration is no longer present (no validation error)

Should the selected drop down entry maybe show a placeholder that indicates that no runtime image configuration is associated with the selected value to distinguish the two?

**Comment by kevin-bates** [comment]
> While the behavior is expected from an implementation point of view 

I'm not so sure I agree with that.  While we want to prevent indirect image references (which would be the case if we persisted the runtime-image metadata _instance_ name), it seems the displayed values should match between the _hover display_ and the _edit display_.  As a result, I recommend we always show the actual image name field value (once set).

I also wonder if we shouldn't present a warning message when updating the image **on a previously saved** pipeline.  (Not sure if that can be detected, but I don't think you'd want a warning if you're simply updating the field in the same edit session.)  The warning message would be something to the effect of _"Warning: Changing the runtime image may change the results of this node."_

**Comment by ptitzler** [comment]
Maybe display both, e.g. `org/image:tag (configuration name)` or something similar. If we only display one or the other, it's not necessarily obvious which one it is. I'm primarily bringing this up because I assume this will happen fairly frequently and surface in our examples, which are targeted at beginners.

**Comment by kevin-bates** [comment]
Yeah, I was afraid that might confuse things since the parenthesized value is not always determinable - unless we persist both - in which case, it's associated runtime instance might not exist, then what?  It's probably worth a further discussion but I think persisting the image name (vs. the instance name) is both less confusing and more repeatable.

**Comment by lresende** [comment]
This is a known issue described in https://github.com/elyra-ai/elyra/issues/497. We have decided, based on the discussion on that issue, to error on the side of reproducibility and require that the user explicitly update to a newer version of the runtime image, while still allowing the pipeline to be run with the old image tag.

**Comment by kevin-bates** [comment]
The issue Patrick is raising is that the 'edit properties' dialog doesn't show the image name when the image name has become _detached_ from its original runtime-image instance.  This results in an ambiguity between nodes that have no image selected and those that have a "detached" image selected.

Since we already show the runtime-image instance name when there *is* a match with a runtime-image instance (and I wasn't aware of that "matching" functionality until now), it seems like we should simply show the image name when no match to a runtime-image can be found. Not sure if there should be an indication, in this case, that the image name is not associated with any runtime-image instance, but that would disambiguate the two scenarios.

**Comment by lresende** [comment]
The match/display is not done by us, but by the control. Should we stop using a `display title` for the image and using the docker image name/tag? Having said that, the current behavior does signalize to the user that he should choose another image based on the current/updated list of images, which if we look into the notebook, it's the same behavior when you have a saved kernel that is not available, it prompts the user to select from the current possibilities.

**Comment by kevin-bates** [comment]
I don't understand your response.  Why should the user have to change the image that works perfectly well (otherwise they'd change the image)?  The issue is that when there's no match (which is out of control), we don't seed the dialog with the image name that **is in use**.  That is misleading.    

Just because we don't have control over the match/display, we do have control over the set to match from.  So it seems we could fabricate "virtual" runtime-image entries that map 'image name' to 'image name' rather than 'image name' to 'runtime-image instance name' in cases where the configured image name that exists in the pipeline is now NOT in the list of runtime-image instances.

Displaying blank for something that is not blank is a bug.
